server/src/main/java/org/apache/iotdb/db/metadata/LocalConfigManager.java

```diff
@@ -289,6 +261,49 @@ private void ensureStorageGroup(PartialPath path, boolean shouldAllocateSchemaRe
     }
   }
 
+  /**
+   * Delete storage groups of given paths from MTree.
+   *
+   * @param storageGroups list of paths to be deleted.
+   */
+  public void deleteStorageGroups(List<PartialPath> storageGroups) throws MetadataException {
+    for (PartialPath storageGroup : storageGroups) {
+      deleteStorageGroup(storageGroup);
+    }
+  }
+
+  private void deleteStorageGroup(PartialPath storageGroup) throws MetadataException {
+    deleteSchemaRegionsInStorageGroup(
```

Comment:
If a `storageGroup` has five `schemaRegions` and failed to delete the third one（In distributed scenarios, the health status of different nodes may be inconsistent, such as disks）, does the user still think the `storage group` is available（because the deletion failed）?  

Maybe we need to fix consistency instead of just fixing a corner case.